content/posts/2025-gsoc-clang-doc.md

```diff
@@ -0,0 +1,275 @@
+- [integrate JSON as the source for Mustache templates](https://github.com/llvm/llvm-project/pull/149589)
+- [separate comments into categories](https://github.com/llvm/llvm-project/pull/149590)
+- [enable comments in class templates](https://github.com/llvm/llvm-project/pull/149848)
+- [remove nesting of text comments inside paragraphs](https://github.com/llvm/llvm-project/pull/150451)
+- [generate comments for functions](https://github.com/llvm/llvm-project/pull/150570)
+- [add param comments to comment template](https://github.com/llvm/llvm-project/pull/150571)
+- [add return comments to comment template](https://github.com/llvm/llvm-project/pull/150647)
+- [add code comments to comment template](https://github.com/llvm/llvm-project/pull/150648)
+
+# Markdown
+Markdown was the most speculative aspect of the project.
+It wasn't clear whether we'd try to integrate a solution into Clang itself or whether we'd keep it in clang-tools-extra.
+
+## A JavaScript Solution
+The first option I explored was suggested by my mentor, which was a JavaScript library called [Markdown-Tag](https://github.com/MarketingPipeline/Markdown-Tag)
+This would've been really convenient since all it requires is an HTML tag to enable rendering, so any comment text in a template can be easily rendered.
+Unfortunately, it requires all HTML to be sanitized, which defeats the purpose of a ready-made solution for us.
+We would have to parse any potential HTML in comments anyways.
+
+## A Parser Solution
+Without an out-of-the-box solution, we were left with implementing our own parser.
+When I considered this in my proposal, I knew an in-tree parser would want to conform to the simplest possible standard.
+Markdown has no official standard, so I opted for CommonMark conformance.
+
+The summer ended without a complete solution since a couple weeks were spent researching whether or not this could be integrated directly in the Clang comment parser or whether we'd need to build our own solution or not.
+You can see my initial draft [here](https://github.com/llvm/llvm-project/pull/155887).
+I plan to continue working on this parser and landing it in Clang-Doc.
+
+# Refactors, Name Mangling, and More!
+During my summer, I would stumble into places where I would think "This could be better" and my mentors usually agreed.
+Thus, there were a few patches where I dedicated time to general refactors to improve code reuse and hopefully make the lives of future contributors much easier than what I had to go through.
+In fact, one of my best refactors was of the JSON generator that I wrote, which my mentor noted had a lot of areas for great code reuse.
+Future me was extremely thankful for the easy-to-use functions I had added.
+I also refactored some of the bitcode reader/writer code so that less copy-pasting would be involved in the future.
+
+Another significant feature that I hadn't planned was name mangling.
```

Comment:
The problem you were solving w/ name mangling was the lack of unique file names.